src/modules/position-creator/services/position.creator.transaction.ts

```diff
@@ -75,53 +77,52 @@ export class PositionCreatorTransactionService {
                 swapRoutes[swapRoutes.length - 1],
             );
 
-        const contract = lockEpochs
-            ? await this.mxProxy.getLockedTokenPositionCreatorContract()
-            : await this.mxProxy.getPostitionCreatorContract();
-
-        let interaction: Interaction;
-
-        if (lockEpochs) {
-            interaction = contract.methodsExplicit.createPairPosFromSingleToken(
-                [
-                    new U64Value(new BigNumber(lockEpochs)),
-                    new BigUIntValue(amount0Min),
-                    new BigUIntValue(amount1Min),
-                    ...swapRouteArgs,
-                ],
-            );
-        } else {
-            interaction = contract.methodsExplicit.createLpPosFromSingleToken([
-                new AddressValue(Address.fromBech32(pairAddress)),
-                new BigUIntValue(amount0Min),
-                new BigUIntValue(amount1Min),
-                ...swapRouteArgs,
-            ]);
-        }
-
         const gasLimit =
             gasConfig.positionCreator.singleToken.liquidityPosition +
             gasConfig.pairs.addLiquidity +
             gasConfig.pairs.swapTokensFixedInput.withFeeSwap *
                 swapRoutes[0].pairs.length;
 
-        interaction = interaction
-            .withSender(Address.fromBech32(sender))
-            .withGasLimit(gasLimit)
-            .withChainID(mxConfig.chainID);
+        const transactionOptions = new TransactionOptions({
+            sender: sender,
+            chainID: mxConfig.chainID,
+            gasLimit: gasLimit,
+            function: 'createLpPosFromSingleToken',
+            arguments: [
+                new AddressValue(Address.newFromBech32(pairAddress)),
+                new BigUIntValue(amount0Min),
+                new BigUIntValue(amount1Min),
+                VariadicValue.fromItems(...swapRouteArgs),
+            ],
+        });
 
         if (payment.tokenIdentifier === mxConfig.EGLDIdentifier) {
-            interaction = interaction.withValue(new BigNumber(payment.amount));
+            transactionOptions.nativeTransferAmount = payment.amount;
         } else {
-            interaction = interaction.withSingleESDTTransfer(
-                TokenTransfer.fungibleFromBigInteger(
-                    payment.tokenIdentifier,
-                    new BigNumber(payment.amount),
-                ),
+            transactionOptions.tokenTransfers = [
+                new TokenTransfer({
+                    token: new Token({
+                        identifier: payment.tokenIdentifier,
+                    }),
+                    amount: BigInt(payment.amount),
+                }),
+            ];
+        }
+
+        if (lockEpochs) {
```

Comment:
add in this if block the return for transaction generated for locked token position creator. No need for a second check below

Reply:
done